common/session/class.BasicSession.php

```diff
@@ -165,4 +168,20 @@ public function setServiceLocator(ServiceLocatorInterface $serviceLocator)
         }
         return $this->setOriginalServiceLocator($serviceLocator);
     }
+
+    public function withContext(SessionContext $context): self
+    {
+        $this->contexts[] = $context;
+        return $this;
+    }
+
+    public function getContexts($class = null): array
+    {
+        $contexts = $this->contexts;
+        if ($class != null) {
+            $contexts = array_filter($contexts, function($element) use ($class) {return $element instanceof $class;});
+        }
+        return $contexts;
```

Comment:
I would use **early return** as it is recommended to avoid usage of `else` or unnecessary pre-assignments (https://williamdurand.fr/2013/06/03/object-calisthenics/#2-dont-use-the-else-keyword).

 ```suggestion
        if ($class === null) {
            return $this->contexts;
        }
        
        return array_filter(
            $this->contexts, 
            function($element) use ($class) {
                return $element instanceof $class;
            }
        );
```
You can avoid creating another variable unnecessarily. I also think we should indent anonymous functions for better readability.

Reply:
While in general the Object Calisthenics are a good read, I strongly disagree with the early return. While early returns are acceptable as part of guards, they should be avoided as they are easy to miss and make the flow of a function harder to understand.

The article in question was about avoiding **ELSE** statements and even suggested to introduce a variable in order to make your return statement parametrizable, as I did.

Reply:
I am recommending object-calisthenics cause they are well know and respected recommendations. I agree with early return, since it will quickly avoid me to continue reading the code when I reached the point I was looking for. It saves time and it is more fluid. 

I would at least consider indenting the anonymous function to make code easier to be read by other colleagues.